build/cmake/CMakeModules/AddZstdCompilationFlags.cmake

```diff
@@ -21,7 +21,10 @@ function(EnableCompilerFlag _flag _C _CXX)
 endfunction()
 
 macro(ADD_ZSTD_COMPILATION_FLAGS)
-    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang" OR MINGW) #Not only UNIX but also WIN32 for MinGW
+    #Not only UNIX but also WIN32 for MinGW
+    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR
+        (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT MSVC) OR
```

Comment:
If only `-Wall` is the problem, could you only turn off `-Wall` for clang-cl. E.g. leave this if as-is and do:

```cmake
if (NOT (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND MSVC))
    EnableCompilerFlag("-Wall" true true)
endif ()